jerry-core/vm/vm.c

```diff
@@ -1948,8 +1948,8 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_ITERATOR_STEP:
         {
-          const int8_t index = (opcode == CBC_EXT_ITERATOR_STEP) ? -1 : -3;
-          result = ecma_op_iterator_step (stack_top_p[index]);
+          const uint8_t index = (uint8_t) (1 + (opcode - CBC_EXT_ITERATOR_STEP));
```

Comment:
Is there a valid range of indexes we support? If so would it be possible to have an assert for it?

Reply:
Yes, there is. Good idea, I'll add an assert for it.